tarantool_test.go

```diff
@@ -3972,6 +3972,104 @@ func TestConnect_context_cancel(t *testing.T) {
 	}
 }
 
+// A dialer that rejects the first few connection requests.
+type mockSlowDialer struct {
+	counter  *int
+	original NetDialer
+}
+
+func (m mockSlowDialer) Dial(ctx context.Context, opts DialOpts) (Conn, error) {
+	*m.counter++
+	if *m.counter < 10 {
+		return nil, fmt.Errorf("Too early: %v", *m.counter)
+	}
+	return m.original.Dial(ctx, opts)
+}
+
+func TestConnectIsBlocked(t *testing.T) {
+	const server = "127.0.0.1:3015"
+	testDialer := dialer
+	testDialer.Address = server
+
+	inst, err := test_helpers.StartTarantool(test_helpers.StartOpts{
+		Dialer:       testDialer,
+		InitScript:   "config.lua",
+		Listen:       server,
+		WaitStart:    100 * time.Millisecond,
+		ConnectRetry: 10,
+		RetryTimeout: 500 * time.Millisecond,
+	})
+	defer test_helpers.StopTarantoolWithCleanup(inst)
+	if err != nil {
+		t.Fatalf("Unable to start Tarantool: %s", err)
+	}
+
+	var counter int
+	mockDialer := mockSlowDialer{original: testDialer, counter: &counter}
+	ctx, cancel := context.WithTimeout(context.TODO(), 10*time.Second)
+	defer cancel()
+
+	reconnectOpts := opts
+	reconnectOpts.Reconnect = 100 * time.Millisecond
+	reconnectOpts.MaxReconnects = 100
+	conn, err := Connect(ctx, mockDialer, reconnectOpts)
+	if err != nil {
+		t.Fatalf("Connection was not established: %v", err)
+	}
+
+	conn.Close()
+
+	if counter < 10 {
+		t.Fatalf("Expected no less than 10, got %d", counter)
+	}
+}
+
+func TestConnectIsBlockedUntilContextExpires(t *testing.T) {
+	const server = "127.0.0.1:3015"
+
+	testDialer := dialer
+	testDialer.Address = server
+
+	ctx, cancel := test_helpers.GetConnectContext()
+	defer cancel()
+
+	reconnectOpts := opts
+	reconnectOpts.Reconnect = 100 * time.Millisecond
+	reconnectOpts.MaxReconnects = 100
+	_, err := Connect(ctx, testDialer, reconnectOpts)
+	if err == nil {
+		t.Fatal("Connection was unexpectedly established.")
+	}
+
+	exp := "failed to dial: dial tcp 127.0.0.1:3015: i/o timeout"
+	if err.Error() != exp {
+		t.Fatalf("Expected '%s', got '%v'", exp, err)
+	}
```

Comment:
```suggestion
	assert.EqualError(t, err, exp)
```
Maybe we can use this? And in all other places

Reply:
Fixed:
<details>

```diff
diff --git a/tarantool_test.go b/tarantool_test.go
index 741e0a4..6a44dc7 100644
--- a/tarantool_test.go
+++ b/tarantool_test.go
@@ -4013,15 +4013,9 @@ func TestConnectIsBlocked(t *testing.T) {
        reconnectOpts.Reconnect = 100 * time.Millisecond
        reconnectOpts.MaxReconnects = 100
        conn, err := Connect(ctx, mockDialer, reconnectOpts)
-       if err != nil {
-               t.Fatalf("Connection was not established: %v", err)
-       }
-
-       conn.Close()
-
-       if counter < 10 {
-               t.Fatalf("Expected no less than 10, got %d", counter)
-       }
+       defer conn.Close()
+       assert.Nil(t, err)
+       assert.GreaterOrEqual(t, counter, 10)
 }
 
 func TestConnectIsBlockedUntilContextExpires(t *testing.T) {
@@ -4037,14 +4031,8 @@ func TestConnectIsBlockedUntilContextExpires(t *testing.T) {
        reconnectOpts.Reconnect = 100 * time.Millisecond
        reconnectOpts.MaxReconnects = 100
        _, err := Connect(ctx, testDialer, reconnectOpts)
-       if err == nil {
-               t.Fatal("Connection was unexpectedly established.")
-       }
-
-       exp := "failed to dial: dial tcp 127.0.0.1:3015: i/o timeout"
-       if err.Error() != exp {
-               t.Fatalf("Expected '%s', got '%v'", exp, err)
-       }
+       assert.NotNil(t, err)
+       assert.ErrorContains(t, err, "failed to dial: dial tcp 127.0.0.1:3015: i/o timeout")
 }
 
 func TestConnectIsUnblockedAfterMaxAttempts(t *testing.T) {
@@ -4060,14 +4048,8 @@ func TestConnectIsUnblockedAfterMaxAttempts(t *testing.T) {
        reconnectOpts.Reconnect = 100 * time.Millisecond
        reconnectOpts.MaxReconnects = 1
        _, err := Connect(ctx, testDialer, reconnectOpts)
-       if err == nil {
-               t.Fatal("Connection was unexpectedly established.")
-       }
-
-       exp := "last reconnect failed"
-       if !strings.Contains(err.Error(), exp) {
-               t.Fatalf("Expected '%s', got '%v'", exp, err)
-       }
+       assert.NotNil(t, err)
+       assert.ErrorContains(t, err, "last reconnect failed")
 }
 
```
</details>